Index: e2fsprogs-1.41.12/misc/mke2fs.c
===================================================================
--- e2fsprogs-1.41.12.orig/misc/mke2fs.c
+++ e2fsprogs-1.41.12/misc/mke2fs.c
@@ -118,7 +118,7 @@ static void usage(void)
 	exit(1);
 }
 
-static int int_log2(int arg)
+static int int_log2(unsigned int arg)
 {
 	int	l = 0;
 
@@ -1276,6 +1276,12 @@ static void PRS(int argc, char *argv[])
 					_("flex_bg size must be a power of 2"));
 				exit(1);
 			}
+			if (flex_bg_size > (unsigned int)INT_MAX + 1) {
+				com_err(program_name, 0,
+					_("flex_bg size must be %u or smaller"),
+					(unsigned int)INT_MAX + 1);
+				exit(1);
+			}
 			break;
 		case 'i':
 			inode_ratio = strtoul(optarg, &tmp, 0);
