From 5648c35ab17270f0e7b87a8c8361c835787f946b Mon Sep 17 00:00:00 2001
From: Valerie Aurora Henson <vaurora@redhat.com>
Date: Sun, 13 Jun 2010 12:00:00 -0400
Subject: [PATCH] badblocks: Use ext2fs_get_device_size2()

Signed-off-by: Valerie Aurora Henson <vaurora@redhat.com>
Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
---
 misc/badblocks.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

Index: e2fsprogs-1.41.12/misc/badblocks.c
===================================================================
--- e2fsprogs-1.41.12.orig/misc/badblocks.c
+++ e2fsprogs-1.41.12/misc/badblocks.c
@@ -981,7 +981,7 @@ int main (int argc, char ** argv)
 	FILE * in = NULL;
 	int block_size = 1024;
 	unsigned int blocks_at_once = 64;
-	blk_t last_block, first_block;
+	blk64_t last_block, first_block;
 	int num_passes = 0;
 	int passes_clean = 0;
 	int dev;
@@ -1116,7 +1116,7 @@ int main (int argc, char ** argv)
 		usage();
 	device_name = argv[optind++];
 	if (optind > argc - 1) {
-		errcode = ext2fs_get_device_size(device_name,
+		errcode = ext2fs_get_device_size2(device_name,
 						 block_size,
 						 &last_block);
 		if (errcode == EXT2_ET_UNIMPLEMENTED) {
